src/tools/schema.js

```diff
@@ -46,23 +41,59 @@ const convertResourceDos = (appRaw) => {
       create.key = `${resource.key}Create`;
       create.noun = resource.noun;
       create.operation.resource = resource.key;
-      create.operation.outputFields = create.operation.outputFields || resource.outputFields;
       creates[create.key] = create;
     }
 
     // TODO: search or create?
   });
 
-  return dataTools.deepCopy({ triggers, searches, creates });
+  return { triggers, searches, creates };
+};
+
+/* When a trigger/search/create (action) links to a resource, we walk up to
+ * the resource and copy missing properties from resource to the action.
+ */
+const copyPropertiesFromResource = (type, action, appRaw) => {
+  if (appRaw.resources && action.operation && appRaw.resources[action.operation.resource]) {
+    const copyableProperties = ['outputFields', 'sample'];
+    const resource = appRaw.resources[action.operation.resource];
+
+    if (type === 'trigger' && action.operation.type === 'hook') {
+      if (resource.list && resource.list.operation && resource.list.operation.perform) {
```

Comment:
instead of this long line we can use `lodash` and `if (_.get(resource, 'list.operation.perform')) {` which will return undefined if any of the properties in that "path" don't exist.

Same for similar code below.